lib/plenario_web/controllers/web/page_controller.ex

```diff
@@ -9,77 +9,196 @@ defmodule PlenarioWeb.Web.PageController do
+
+  # first load
+
   def explorer(conn, params) when params == %{} do
-    render_explorer(conn, nil, "[41.9, -87.7]", 10, nil, "", "")
+    render conn, "explorer.html",
+      results: nil,
+      granularity: @default_granularity,
+      map_center: @default_center,
+      map_zoom: @default_zoom,
+      bbox: nil,
+      starts: nil,
+      ends: nil
   end
 
+  # user submits form
+
   def explorer(conn, params) do
-    zoom = Map.get(params, "zoom")
-    coords = Map.get(params, "coords")
-    starts = Map.get(params, "starting_on")
-    ends = Map.get(params, "ending_on")
+    granularity = parse_granularity(params)
+    zoom = parse_zoom(params)
+
+    starts = parse_date(params, "starting_on")
+    ends = parse_date(params, "ending_on")
+    time_range = make_time_range(starts, ends)
+
+    conn =
+      case is_nil(time_range) do
+        true ->
+          conn
+          |> put_status(:bad_request)
+          |> put_flash(:error, "You must select a time range with a starting date less than the ending date.")
+
+        false ->
+          conn
+      end
 
-    {startdt, conn} = parse_date(conn, starts)
-    {enddt, conn} = parse_date(conn, ends)
+    bbox = parse_coords(params)
+    center = find_center(bbox)
 
-    do_explorer(zoom, coords, startdt, enddt, conn)
-  end
+    conn =
+      case is_nil(bbox) do
+        true ->
+          existing_message =
+            case get_flash(conn)[:error] do
+              nil -> ""
+              value -> value
+            end
 
-  defp parse_date(conn, date_string) do
-    case NaiveDateTime.from_iso8601("#{date_string}T00:00:00.0") do
-      {:error, _} ->
-        {nil, put_flash(conn, :error,
-        "Invalid date #{inspect(date_string)}")}
-      {_, datetime} ->
-        {datetime, conn}
-    end
-  end
+          message =
+            existing_message <>
+            "You must select an area on the map to search data sets."
+
+          conn
+          |> put_status(:bad_request)
+          |> put_flash(:error, message)
+
+        false ->
+          conn
+      end
+
+    results =
+      case is_nil(bbox) do
+        true ->
+          nil
+
+        false ->
+          Plenario.search_data_sets(bbox, time_range)
+      end
 
-  defp do_explorer(nil, nil, nil, nil, conn) do
-    render_explorer(conn, nil, "[41.9, -87.7]", 10, nil, "", "")
+    coords =
+      case is_nil(bbox) do
+        true ->
+          nil
+
+        false ->
+          bbox.coordinates |> Poison.encode!()
+      end
+
+    render conn, "explorer.html",
+      results: results,
+      granularity: granularity,
+      map_center: center,
+      map_zoom: zoom,
+      bbox: coords,
+      starts: starts,
+      ends: ends
   end
 
-  defp do_explorer(_, _, _, _, conn = %Plug.Conn{private: %{:phoenix_flash => %{"error" => _}}}) do
-    render_explorer(conn, nil, "[41.9, -87.7]", 10, nil, "", "")
+  defp parse_granularity(params), do: Map.get(params, "granularity", @default_granularity)
```

Comment:
I like "granularity"; it reads better than our old clumsy "aggregate by" and the like.